paimon-core/src/main/java/org/apache/paimon/utils/FileStorePathFactory.java

```diff
@@ -38,7 +38,10 @@ public class FileStorePathFactory {
 
     public static final String BUCKET_PATH_PREFIX = "bucket-";
 
+    // this is the table schema root path
     private final Path root;
+    // this is the table data root path
+    private final Path dataRoot;
```

Comment:
May be it is better named externalRoot?

Reply:
This is the current location where data is written, but it may not necessarily be the external path, it could also be the warehouse root, so it is called `dataRoot` here.  It is the same as `CoreOptions.TABLE_DATA_PATH`, as it description "The data file path of this table in the filesystem. if `data-file.external-path` is not set, it will be same with `PATH.key`"

Reply:
ok